python/ray/autoscaler/_private/gcp/node.py

```diff
@@ -0,0 +1,549 @@
+from typing import List, Optional, Tuple, Union
```

Comment:
add top-file comments about why this abstraction is needed, and how should this be maintained in the future if GCP updates its APIs?